src/jdk.jpackage/share/native/common/FileUtils.h

```diff
@@ -72,6 +72,9 @@ namespace FileUtils {
      */
     tstring replaceSuffix(const tstring& path, const tstring& suffix=tstring());
 
+    // extract the name from the launcher path
+    tstring extractName(const tstring& path);
```

Comment:
Function name seems to be misleading. 
If it specifically designed to be applied to executables, I'd name it `stripExecutableSuffix()`. On Unix implementation would return passed in string as is without calling `basename()` and on Windows would just strip ".exe" suffix not calling `basename()` either.
The use would be: `FileUtils::mkpath() << appDirPath << (FileUtils::stripExecutableSuffix(FileUtils::basename(launcherPath)) + _T(".cfg"));`

Reply:
makes sense - updated as requested